### DISCUSSION TITLE
2D Object Detection: Making CL useful towards autonomous vehicles

### DISCUSSION THREAD

**Discussion by Mattdl** [ContinualAI/avalanche#517] [Feature Request]
Hi all,

I was wondering how we could extend Avalanche to the 2d object detection task.
Right now the focus is supervised classification tasks, but I think the Strategies generalize to this task as well.

Probably the main implementation efforts would be new metrics (recal/precision -> Which can all be derived from the confusion matrix), and some 2d object detection BenchMarks.

I think this is a very important direction to put Avalanche also on the map for a lot of companies working on autonomous vehicles etc. 

Therefore, I was hoping to start a discussion on this: What do you think are the main implementation efforts required?

**Comment by AntonioCarta** [comment]
I agree with you. Benchmarks should be quite general right now. Strategies can be easily generalized but will probably need some changes. Metrics need some work to make them parametric (e.g. allow slightly different accuracy computations) and to easily create new MetricPlugins.

I think we need to identify a couple of benchmarks that are of general interest and encapsulate all these pain points so that we can use them as testbeds to improve strategies and metrics. @Mattdl do you have some specific benchmarks in mind?

**Reply by Mattdl** [reply]
I think PASCAL VOC is a good starting point as it's rather small-scale. If the basics are implemented, we can proceed further on the standard MSCOCO. 

I'm not too familiar with the object detection setup myself, but maybe we can try to encourage some authors with experience on this regard to contribute. e.g. I found this recent paper combining detection with CL: https://arxiv.org/pdf/2002.05347.pdf
